### ISSUE TITLE
[Bug]: flatpak update fails if using DISPLAY="" instead of unset DISPLAY

### ISSUE THREAD

**Issue by GloriousEggroll** [flatpak/flatpak#5702]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

1.15.4

### What Linux distribution are you using?

Other (specify below)

### Linux distribution version

Nobara Linux 39

### What architecture are you using?
x86_64

### How to reproduce

In Nobara we have an updater script that has both a GUI option and a CLI option.

If a display is detected, it runs the updater with the GUI unless 'cli' is explicitly added as a launch option.

When 'cli' is added as a launch option, we used `DISPLAY=""` to empty the display envvar. 

When doing this,  we noticed `flatpak update` commands inside the script would fail with logs like so:
```
QUESTION: Flatpak has been detected! Would like to update all Flatpaks on your system? [y/n] y
Updating appstream data for remote flathub
Error updating: Error updating appstream2: No such ref 'appstream2/x86_64' in remote flathub; Error updating appstream: No such ref 'appstream/x86_64' in remote flathub
Looking for updates…

Nothing to do.
Updating appstream data for remote flathub
Error updating: Error updating appstream2: No such ref 'appstream2/x86_64' in remote flathub; Error updating appstream: No such ref 'appstream/x86_64' in remote flathub
Updating appstream data for user remote flathub
Error updating: Error updating appstream2: No such ref 'appstream2/x86_64' in remote flathub; Error updating appstream: No such ref 'appstream/x86_64' in remote flathub
Updating appstream data for user remote ulwgl
Error updating: Error updating appstream2: No such ref 'appstream2/x86_64' in remote ulwgl; Error updating appstream: No such ref 'appstream/x86_64' in remote ulwgl
Looking for updates…
F: Warning: Treating remote fetch error as non-fatal since runtime/org.gtk.Gtk3theme.Breeze/x86_64/3.22 is already installed: No such ref 'runtime/org.gtk.Gtk3theme.Breeze/x86_64/3.22' in remote flathub
F: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform/x86_64/23.08 is already installed: No such ref 'runtime/org.freedesktop.Platform/x86_64/23.08' in remote flathub
F: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform.openh264/x86_64/2.2.0 is already installed: No such ref 'runtime/org.freedesktop.Platform.openh264/x86_64/2.2.0' in remote flathub
F: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform.Locale/x86_64/23.08 is already installed: No such ref 'runtime/org.freedesktop.Platform.Locale/x86_64/23.08' in remote flathub
F: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform.GL.default/x86_64/23.08-extra is already installed: No such ref 'runtime/org.freedesktop.Platform.GL.default/x86_64/23.08-extra' in remote flathub
F: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform.GL.default/x86_64/23.08 is already installed: No such ref 'runtime/org.freedesktop.Platform.GL.default/x86_64/23.08' in remote flathub

Nothing to do.
```

We managed to fix the issue by using `unset DISPLAY` instead of `DISPLAY=""`,  however felt it might be worth reporting as it's quite possible someone else might hit the same issue and I don't think flatpak's intention is to fail on an empty string.

### Expected Behavior

`flatpak update` should run as normally expected if `DISPLAY` envvar string is empty or unset, not unset only.

### Actual Behavior

`flatpak update` trips over an empty `DISPLAY` string

**Comment by chrisawi** [comment]
```
$ G_MESSAGES_DEBUG=all DISPLAY="" flatpak remote-ls flathub
[...]
flatpak-INFO: 22:09:05.362: Fetching summary index file for remote ‘flathub’
flatpak-INFO: 22:09:05.362: Loading https://dl.flathub.org/repo/summary.idx using curl
flatpak-INFO: 22:09:05.444: Received 10001 bytes
flatpak-INFO: 22:09:05.444: Loading https://dl.flathub.org/repo/summaries/d1083f0d5b6f13824e4dc3e78121335b5cb9aed0c1d69f1df4775552548da896.idx.sig using curl
flatpak-INFO: 22:09:05.467: Received 592 bytes
(flatpak remote-ls:3331611): OSTree-DEBUG: 22:09:05.468: Adding GPG keyring data from flathub.trustedkeys.gpg to verifier
flatpak-INFO: 22:09:05.496: Failed to download optional summary index: GPG: Unable to complete signature verification: GnuPG: General error
error: Unable to load summary from remote flathub: GPG: Unable to complete signature verification: GnuPG: General error
```

Running it again with `GPGME_DEBUG=9` reveals: 

```
gpgme[3332282.32d8ba]     _gpgme_io_read: check: gpg: missing argument for option "--display="<LF>
```

So this is a gpgme bug (via ostree).

**Comment by smcv** [comment]
If you consider this to be a gpgme bug, then please report it to gpgme (although I'm not convinced that it's a bug at all, because empty and unset are distinct states for environment variables, and the results of doing one when you meant the other are unpredictable and context-dependent). If this is indeed a bug, then I don't intend to work around it in Flatpak.

**Comment by chrisawi** [comment]
In my mind, the (minor) bug in gpgme is that it constructs an invalid `gpg` command line. I have no opinion on whether it should treat empty as unset, but if it doesn't, it should explicitly error out or find some way to pass the empty string to gpg.

I agree there's nothing to do in flatpak, except maybe stop `flatpak update` from swallowing errors, but there are [other issues](https://github.com/flatpak/flatpak/issues?q=is%3Aissue+%22treating+error+as+non-fatal%22+is%3Aopen) for that.

**Comment by chrisawi** [comment]
Closing as not a flatpak bug. https://github.com/flatpak/flatpak/issues/5134 (among others) tracks `flatpak update` 's opaque output.

**Comment by ryonakano** [comment]
Leaving a note that the empty DISPLAY issue should be fixed in gpgme 2.1.0.

- Upstream bug tracker: https://dev.gnupg.org/T7919.html
- Upstream fix: https://github.com/gpg/gpgme/commit/94fa3c2b25a1979bb838c7d9f9f74375072f4efe
